### ISSUE TITLE
Error while self hosting: Undefined array key 0 smarty_streets.php line 69

### ISSUE THREAD

**Issue by StarScream159** [beestat/app#391]
**Describe the bug**
A new fresh install for self hosting. After setup and browsing to the app I am redirected to Ecobee for login. Once logged in I am brought back to the app with an API call failed error:

```
{
  "success": false,
  "data": {
    "error_message": "Undefined array key 0",
    "error_code": 2,
    "error_detail": {
      "file": "/opt/beestat/api/smarty_streets.php",
      "line": 69,
```

Var dumping the line above the error I get:
```
array(1) {
  ["errors"]=>
  array(1) {
    [0]=>
    array(2) {
      ["id"]=>
      int(1611079217)
      ["message"]=>
      string(36) "Authentication required (1611079217)"
    }
  }
}
```

To self host do we need to setup "smarty_streets" credentials in the setting.php file? The readme didn't touch on that. My `smarty_streets_auth_id` and `smarty_streets_auth_token` are blank strings in setting.php.

**Version Information**
OS: AlmaLinux release 9.1 (Lime Lynx)
MySQL: mysql  Ver 8.0.30 for Linux on x86_64 (Source distribution)
PHP: PHP 8.1.17 (cli) (built: Mar 14 2023 19:50:20) (NTS gcc x86_64)
beestat version:
```
[root@localhost beestat]# git reset --hard
HEAD is now at b9b3949 Debug/fix for #387
[root@localhost beestat]# git status
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean
[root@localhost beestat]#
```

Thanks for all the help.

**Comment by StarScream159** [comment]
Yes, that worked. Thank you.

**Comment by ziebelje** [comment]
That's great! Let me know if you have any other problems...glad to see someone got a self-install working!

**Comment by StarScream159** [comment]
Hi @ziebelje. My install has been working great since however suddenly I get an error now when I browse to the instance. Let me know if you'd prefer I open a new issue.

**API call failed: Incorrect datetime value: '2024-03-10 02:00:00' for column 'timestamp' at row 1**

The stacktrace is very long so I've included it as a file.

That looks like a valid datetime to me. If I had to guess this is coming from the Ecobee API?

[stacktrace.txt](https://github.com/beestat/app/files/14872287/stacktrace.txt)

**Comment by StarScream159** [comment]
I was able to fix my issue by adjusting the MYSQL time zone to UTC. I have mysql running on a separate server then php. Mysql was set to use the system time zone and while I still don't want to change the ssytem time zone, I did change mysql's via the conf file my.cnf:
```
[mysqld]
default-time-zone='+00:00'
````
This seems to have worked after restarting mysql. The interface now loads sycned/fetched and everything looks correct.

Sorry for reopening an old issue. Thanks for reading.
